hail/python/hail/experimental/densify.py

```diff
@@ -36,6 +36,7 @@ def densify(sparse_mt):
                     prev_entries[i], t.__entries[i],
                     lambda prev_entry, entry:
                     hl.cond(
+                        (~hl.is_defined(entry)) &
```

Comment:
I agree this is now correct (don't overwrite a defined entry), but we're scanning with `entry` (see line 31), which means the defined entry will overwrite and prematurely truncate the REF block.  I think we should scan with `hl.or_missing(hl.is_defined(entry.END), entry)` (that is, just REF blocks), so defined non-REF blocks don't overwrite REF blocks in the scan.

Then I think the densified output is correct despite the bug in the input.  Good catch, @chrisvittal!

Reply:
I also think that we can stop checking for presentness in `prev_entry.END` as we will have checked for it earlier. See my latest updates.